tests/ArrayAccessTest.php

```diff
@@ -1355,7 +1355,7 @@ function takesList(array $arr) : array {
 
                         return $arr;
                     }',
-                'error_message' => 'LessSpecificReturnStatement',
+                'error_message' => 'InvalidReturnStatement',
```

Comment:
This is correct IMO, but I _only_ think this works in Psalm's type system if we're willing to go further here and say that calling `unset` for any list element would turns it into a (new-in-Psalm) `map-array<int, string>`, which is a subtype of `array<int, string>` and disjoint from `list<string>`

Reply:
Hmm, with this PR psalm correctly infers the type of $arr to be `array<int<1, max>, string>`, which is not a list, how is an InvalidReturnStatement wrong here?

Reply:
Fixed!